### PR TITLE
Updated Dockerfile

### PR DESCRIPTION
**Pull request by koehn**
updated dockerfile to produce smaller image with minimum content/packages installed, and a few extra features (like not running as root); added .dockerignore

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,5 +11,21 @@ RUN make install
...
+RUN mkdir /stage && \
+    cp -R /go/bin \
+       /go/src/app/templates \
+       /go/src/app/static \
+       /go/src/app/schema.sql \
+       /go/src/app/pages \
+       /go/src/app/keys \
+      /stage && ls -l /stage
+
+FROM golang:1.11.2-alpine3.8
```
Comment (jlelse) [inline Dockerfile]:
```suggestion
FROM scratch
```
Can't the final image be based on scratch or alpine? `go` is probably not needed after building.

Comment (koehn) [reply Dockerfile]: Yeah, I didn't understand Go doesn't need anything else at runtime; my ignorance. Updated to use alpine:3.8.

**Comment by thebaer** [comment]
Thanks for the PR @koehn, and @jlelse for looking at this!

@koehn can you check out the agreement in the Contributing guide? If you're good with that, we can move forward. I'd also like to have @BenOvermyer take a look at these changes and see if they match with what he was planning on doing.

**Comment by koehn** [comment]
I’ve looked at the contributors guide and am good with it. Thanks!

**Comment by thebaer** [comment]
Great :+1: specifically, could you sign the agreement mentioned there?

**Comment by koehn** [comment]
Done.

**Comment by BenOvermyer** [comment]
These are exactly the changes I was going to make. Great job on this, @koehn !

**Comment by thebaer** [comment]
Awesome, I'll get this merged in now. Thanks everyone!

Could either of you update the README if anything needs to change there? Please feel free to commit directly to `master` for small updates like that.

**Comment by koehn** [comment]
I'll try to take on the README. I have a few questions:

1. Is there (or can there be) a `writeas` Docker group, under which we can publish images e.g., `writeas/writefreely:latest` or `writeas/writefreely:0.3.0`? I can create one. 
2. Can publishing images become a part of the standard release process, alongside making the zip release?
3. Any issues with me updating the `docker-compose.yml` file to use the `Dockerfile` and/or the published image?

**Comment by thebaer** [comment]
Great. To your questions:

1. Sure, we can set that up. Would that be on something like Docker Hub? Wherever it is, I can create an account and give you access to do everything you need.
2. Yep, happy to include that in the standard release process. Of course, automating this in any way we can will be best. Right now I run `make release` to build a new release, so that'd be the place to start.
3. I don't have any problems with that if it makes setup easier / smoother, but anyone else can feel free to weigh in here.

**Comment by koehn** [comment]
We can also have another docker image for production use that uses the image. The main advantage being that you just need to `docker-compose pull && docker-compose up -d` to upgrade; there's no need to clone the git repo.

**Comment by koehn** [comment]
Not sure if my other message reached you: can you push the 0.4 docker image? Thanks!

**Comment by thebaer** [comment]
Yep, done.

**Pull request merged**